include/velocypack/SharedSlice.h

```diff
@@ -0,0 +1,362 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2020 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Tobias Gödderz
+////////////////////////////////////////////////////////////////////////////////
+
+#ifndef SRC_SHAREDSLICE_H
+#define SRC_SHAREDSLICE_H
+
+#if __cplusplus < 201703L
+#error "This file can only be used with at least C++17. Set CMAKE_CXX_STANDARD=17."
```

Comment:
Is the use of CMake mandatory? Maybe this will come with dds or conan?

Reply:
It was meant to be helpful to the average and innocent user, who will most probably use either the CMakeFiles provided here, or in the arangodb repository.